plugins/tekton/src/components/pipeline-topology/PipelineTaskNode.tsx

```diff
@@ -43,18 +46,39 @@ const PipelineTaskNode = ({
   contextMenuOpen,
   ...rest
 }: PipelineTaskNodeProps) => {
+  const [open, setOpen] = React.useState<boolean>(false);
+  const { watchResourcesData } = React.useContext<TektonResourcesContextData>(
+    TektonResourcesContext,
+  );
   const data = element.getData();
+  const pipelineRun = data.pipelineRun;
   const [hover, hoverRef] = useHover();
   const detailsLevel = useDetailsLevel();
   const isFinallyTask = element.getType() === NodeType.FINALLY_NODE;
 
+  const pods = React.useMemo(
+    () => watchResourcesData?.pods?.data || [],
+    [watchResourcesData],
+  );
+  const taskRuns = React.useMemo(
+    () => watchResourcesData?.taskruns?.data || [],
+    [watchResourcesData],
+  );
```

Comment:
do we need memo here or see some improvement in performance? As there is no computation here